deepchem/models/tensorgraph/tensor_graph.py

```diff
@@ -222,13 +232,31 @@ def create_feed_dict():
             time.sleep(0)
           if n_samples == final_sample[0]:
             break
+
         n_samples += 1
         should_log = (self.tensorboard and
                       n_samples % self.tensorboard_log_frequency == 0)
         fetches = [train_op, loss.out_tensor]
         if should_log:
           fetches.append(self._get_tf("summary_op"))
+
+        run_start = time.time()
+
+        # Commented out but this is really useful for profiling.
```

Comment:
should this be behind a "profile" flag to TG?

Reply:
That would be a really handy feature!

Reply:
I think we need to discuss a little how we want to do this and likely address this in a separate PR.